src/tests/intg/test_files_provider.py

```diff
@@ -294,6 +308,11 @@ def check_group_list(exp_groups_list):
         check_group(exp_group)
 
 
+def assert_user_overriden():
+    ent.assert_passwd_by_name(USER1["name"], OV_USER1)
```

Comment:
There is an issue in nss_wrapper (https://bugzilla.samba.org/show_bug.cgi?id=12883) and nss_wrapper always looks into the files first before using the NSS module. This lets this check fail because the use is found in the file and hence will be returned without overridden values.

I can think of two possible ways to get around this while there is no fix for nss_wrapper. First the fully-qualified name can be used here

    ent.assert_passwd_by_name(USER1["name"]+"@files", OV_USER1)

or go directly to SSSD:

    res, user = sssd_getpwnam_sync(USER1["name"])
    assert res == NssReturnCode.SUCCESS
    assert user == OV_USER1